test/testkit/mocks/servicenamebundle/service_name_bundle.go

```diff
@@ -61,7 +61,7 @@ func K8sObjects(namespace string, signalType telemetrygen.SignalType) []client.O
 			WithLabel("app", AppLabelValue).
 			WithPodSpec(podSpecWithUndefinedService).
 			K8sObject(),
-		kitk8s.NewDeployment(DeploymentName, namespace).WithPodSpec(podSpecWithUndefinedService).K8sObject(),
+		kitk8s.NewDeployment(DeploymentName, namespace).WithPodSpec(podSpecWithUndefinedService).WithPersistentLabel().K8sObject(),
```

Comment:
PersistentLabel is only needed for upgrade tests, so no need to add it here

Reply:
fixed